src/pages/dashboard.tsx

```diff
@@ -1,12 +1,22 @@
-import { Container, Title } from '@mantine/core';
-import UserForm from '../components/UserForm';
+import { Container } from '@mantine/core';
+
+import { Group } from '@mantine/core';
+import NewsComponent from '../components/NewsComponent';
+import TodoList from '../components/TodoList';
 import WeatherComponent from '../components/WeatherComponent';
+
 const Dashboard = () => {
   return (
-    <Container>
-      <Title order={1}>This is the Dashboard page</Title>
-      <WeatherComponent />
-      <UserForm />
+    <Container style={{ display: 'grid', gap: '20px' }}>
+      <Group style={{ gridColumn: '1 / span 3' }}>
+        <WeatherComponent />
+      </Group>
+      <Group style={{ gridColumnStart: 1, gridColumnEnd: 1.5 }}>
```

Comment:
Not really utilising the Mantine component correctly, I'll create an issue to refractor this.